torch/jit/__init__.py

```diff
@@ -369,7 +373,11 @@ def _script_graph(fn, _frames_up=0):
 
 def script(fn, optimize=True, _frames_up=0):
     graph = _script_graph(fn, _frames_up=_frames_up + 1)
-    return functools.wraps(fn)(torch._C.GraphExecutor(graph, optimize))
+    mod = ScriptModule()
+    mod._create_method_from_graph('forward', graph)
+    # Forward docstrings
+    mod.__doc__ = fn.__doc__
+    return mod
```

Comment:
Is there any noticeable difference in cost of calling a `ScriptModule` vs a `GraphExecutor`?

Reply:
I just tried on my mac:

Benchmark code
```
import torch

@torch.jit.script
def a_fn(x, y):
    return x + y

import time

x = torch.rand(3, 3)
y = torch.rand(3, 3)
s = time.time()
for i in range(10000):
    a_fn(x, y)
print('time elapsed:', time.time() - s)
```

Calling a GraphExecutor
```
time elapsed: 0.12578201293945312
```

Calling a ScriptModule
```
time elapsed: 0.26899099349975586
```

Reply:
@apaszke I just pushed an update that reduces the overhead by doing the `forward()` method lookup in C++, thus we don't incur pybind11 overhead twice. New timing (same benchmark):

Before
```
time elapsed: 0.15010666847229004
```

After
```
time elapsed: 0.16915202140808105
```